packages/react-components/src/menu/src/Menu.tsx

```diff
@@ -278,15 +271,14 @@ export function InnerMenu({
                 }}
             >
                 {nodes.map(node => {
-                    switch (node.type) {
-                        case NodeType.item:
-                            return renderOption(node);
-                        case NodeType.section:
-                            return renderSection(node as CollectionSection);
-                        case NodeType.divider:
-                            return renderDivider(node as CollectionDivider);
-                        default:
-                            return null;
+                    if (isItem(node)) {
```

Comment:
Did you do this to prevent a cast? Not sure it really worth it???

Reply:
reverted